release/nightly_tests/dataset/operator_fusion_benchmark.py

```diff
@@ -0,0 +1,182 @@
+    return {
+        "mean_execution_time": np.mean(execution_times),
+        "max_execution_time": np.max(execution_times),
+        "min_execution_time": np.min(execution_times),
+        "std_execution_time": np.std(execution_times),
+    }
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--num-tasks", type=int, default=1)
+    parser.add_argument("--num-blocks-per-task", type=int, default=1024)
+    parser.add_argument("--block-size", type=int, default=8**1024)
+    parser.add_argument("--data-format", type=str, default="simple")
+    parser.add_argument("--num-columns", type=int, default=1)
+    parser.add_argument(
+        "--ops-spec",
+        type=str,
+        default=(
+            '[{"op": "map_batches", "batch_size": 1024, "batch_format": "default"}]'
+        ),
+    )
+    parser.add_argument("--target-max-block-size", type=int, default=None)
+    parser.add_argument("--disable-optimizer", action="store_true", default=False)
+    parser.add_argument("--num-trials", type=int, default=1)
+    args = parser.parse_args()
+
+    # Only allow num_columns > 0 when not using the simple data format.
+    assert args.num_columns == 1 or args.data_format != "simple"
+
+    # Load the ops spec JSON.
+    ops_spec = json.loads(args.ops_spec)
+
+    target_max_block_size = args.target_max_block_size
+    if target_max_block_size is None:
+        target_max_block_size = DEFAULT_TARGET_MAX_BLOCK_SIZE
+
+    print(
+        f"\nRunning zero-copy batching benchmark for {args.num_trials} trials:\n"
+        f"num_tasks={args.num_tasks}\nnum_blocks_per_task={args.num_blocks_per_task}\n"
+        f"block_size={args.block_size}\ndata_format={args.data_format}\n"
+        f"num_columns={args.num_columns}\n"
+        f"target_max_block_size={target_max_block_size}\nray_commit={ray.__commit__}\n"
+        f"ops_spec:\n{json.dumps(ops_spec, indent=4)}"
+    )
+
+    ray.init()
+
+    ctx = DatasetContext.get_current()
+    ctx.target_max_block_size = target_max_block_size
+    if args.disable_optimizer:
+        ctx.optimizer_enabled = False
+    else:
+        ctx.optimizer_enabled = True
+    results = []
+    for trial in range(args.num_trials):
+        print(f"\n\nRunning trial {trial}\n")
+        print("\tCreating dataset.\n")
+        start = time.perf_counter()
+        ds = make_ds(
+            args.num_tasks,
+            args.num_blocks_per_task,
+            args.block_size,
+            args.data_format,
+            args.num_columns,
+            ops_spec,
+            target_max_block_size,
+        )
+        print("\tExecuting dataset.\n")
+        execute_ds(ds)
+        execution_time = time.perf_counter() - start
+        trial_results = {"execution_time": execution_time}
+        print(f"\tTrial {trial} done: ", trial_results)
+        results.append(trial_results)
+    result_summary = _summarize_results(results)
+    print("\n\nResults: ", result_summary)
```

Comment:
shall we write the result to JSON file as well? So it can be used in dashboard?
`os.environ.get("TEST_OUTPUT_JSON", "/tmp/result.json")` as other tests.